app/policies/eligibilities/evidence_policy.rb

```diff
@@ -35,25 +35,58 @@ def can_destroy?
     #
     # @note The user is the one who is trying to perform the action. The record_user is the user who owns the record. The record is an instance of Eligibilities::Evidence.
     def allowed_to_modify?
-      role_has_permission_to_modify? || (current_user == associated_user)
+      return false unless individual_market_role_identity_verified?
+      return true if (current_user == associated_user)
+      return false unless role.present?
+      return can_hbx_staff_modify? if role.is_a?(HbxStaffRole)
+      return can_broker_modify? if (has_active_broker_role? || has_active_broker_agency_staff_role?)
+      false
     end
 
-    def role_has_permission_to_modify?
-      role.present? && (can_hbx_staff_modify? || can_broker_modify?)
+    def individual_market_role_identity_verified?
+      return true if (associated_person.resident_role || associated_person.consumer_role&.identity_verified?)
+      false
     end
 
     def can_hbx_staff_modify?
       role.is_a?(HbxStaffRole) && role&.permission&.modify_family
     end
 
     def can_broker_modify?
-      (role.is_a?(::BrokerRole) || role.is_a?(::BrokerAgencyStaffRole)) && broker_agency_profile_matches?
+      (has_active_broker_role? || has_active_broker_agency_staff_role?) && matches_individual_market_broker_account?
```

Comment:
Isn't this a repeat of line 42?

Reply:
line 42 checks if role is a broker, then only it will execute can_broker_modify? method.